docs/modules/ROOT/pages/clc-cpmap.adoc

```diff
@@ -0,0 +1,369 @@
+= clc cpmap
+
+CPMap commands are a group of CPMap operations.
+
+Usage:
+
+[source,bash]
+----
+clc cpmap [command] [flags]
+----
+
+== Commands
+
+* <<clc-cpmap-compare-and-set, clc cpmap compare-and-set>>
+* <<clc-cpmap-delete, clc cpmap delete>>
+* <<clc-cpmap-destroy, clc cpmap destroy>>
+* <<clc-cpmap-get, clc cpmap get>>
+* <<clc-cpmap-put, clc cpmap put>>
+* <<clc-cpmap-remove, clc cpmap remove>>
+* <<clc-cpmap-set, clc cpmap set>>
+
+== clc cpmap compare-and-set
+
+Sets the value for the given key if it has the expected value.
+
+Usage:
+
+[source,bash]
+----
+clc cpmap compare-and-set [key] [expected-value] [new-value] [flags]
+----
+
+Parameters:
+
+[cols="1m,1a,2a,1a"]
+|===
+|Parameter|Required|Description|Default
+
+|`--name`, `-n`
+|Optional
+|Name of the map.
+|`default`
+
+|`key`
+|Required
+|Key of the map entry.
+|N/A
+
+|`expected-value`,
+|Required
+|Value to set for the key.
+|N/A
+
+|`new-value`,
+|Required
+|Value to set for the key.
+|N/A
+
+|`--key-type`, `-k`
+|Optional
+|Data type of the key. One of: `string`, `bool`, `json`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`
+|string
+
+|`--value-type`, `-v`
+|Optional
+|Data type of the value. One of: `string`, `bool`, `json`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`
+|string
+
+|===
+
+.Global parameters
+[%collapsible]
+====
+include::partial$global-parameters.adoc[]
+====
+
+== clc cpmap get
+
+Gets a value from the given CPMap.
+
+Usage:
+
+[source,bash]
+----
+clc cpmap get [key] [flags]
+----
+
+Parameters:
+
+[cols="1m,1a,2a,1a"]
+|===
+|Parameter|Required|Description|Default
+
+|`--name`, `-n`
+|Optional
+|Name of the CPMap.
+|`default`
+
+|`key`
+|Required
+|Key of the CPMap entry.
+|N/A
+
+|`--key-type`, `-k`
+|Optional
+|Data type of the key. One of: `string`, `bool`, `json`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`.
+|string
+
+|===
+
+.Global parameters
+[%collapsible]
+====
+include::partial$global-parameters.adoc[]
+====
+
+Example:
+
+[source,bash]
+----
+clc cpmap get -n yearbook -k i16 2012
+----
+
+== clc cpmap set
+
+Sets a value in the given CPMap.
+
+Usage:
+
+[source,bash]
+----
+clc cpmap set [key] [value] [flags]
+----
+
+Parameters:
+
+[cols="1m,1a,2a,1a"]
+|===
+|Parameter|Required|Description|Default
+
+|`--name`, `-n`
+|Optional
+|Name of the CPMap.
+|`default`
+
+|`key`
+|Required
+|Key of the CPMap entry.
+|N/A
+
+|`value`,
+|Required
+|Value to set for the key.
+|N/A
+
+|`--key-type`, `-k`
+|Optional
+|Data type of the key. One of: `string`, `bool`, `json`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`
+|string
+
+|`--value-type`, `-v`
+|Optional
+|Data type of the value. One of: `string`, `bool`, `json`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`
+|string
+
+|===
+
+.Global parameters
+[%collapsible]
+====
+include::partial$global-parameters.adoc[]
+====
+
+Example:
+
+[source,bash]
+----
+clc cpmap -n my-cpmap set -k string hello -v f32 19.94
+----
+
+
+== clc cpmap put
+
+Put sthe given value in the given CPMap and return sthe old value.
```

Comment:
```suggestion
Puts the given value in the given CPMap and returns the old value.
```

Reply:
Thanks Rebekah,
I've fixed the typo and added the reference for CPMap advanced script functions at:
https://github.com/hazelcast/clc-docs/pull/19/commits/298d2a3c821552757a99b36086dac604e0f36b43

Reply:
All changes look good. Thanks Yuce